app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,8 @@
+class CategoriesController < ApplicationController
+  def show
+    @category = Category.find_by id: params[:id]
+    redirect_to root_path unless @category
```

Comment:
```suggestion
    return redirect_to root_path unless @category
```
Sao không return luôn như này nhể

Reply:
Ơ... em chưa nghĩ đến :((